### ISSUE TITLE
CheckoutAPIService.createToken result block never called

### ISSUE THREAD

**Issue by ajf-mr** [checkout/frames-ios#542]
**Describe the bug**
After migrating to v4.3.6, the CheckoutAPIService.createToken result block doesn't get called when using Apple Pay.

**To Reproduce**
1. Get a `PKPayment` from `PKPaymentAuthorizationControllerDelegate didAuthorizePayment` 
2. Initialize `CheckoutAPIService` instance
3. Call CheckoutAPIService.createToken
4. Result block never gets called thus resulting in 'Payment not Completed' in Apple Pay

**Expected behavior**
CheckoutAPIService.createToken result block should be called (failure or success)

**Smartphone (please complete the following information):**
 - Device: iPhone XS
 - OS: iOS 17.5.1

**Additional context**
- Can reproduce issue in both sandbox and production environments
- Can reproduce issue in other devices
- Same code works well with Frames v4.3.2

**PKPaymentAuthorizationControllerDelegate**
```
func paymentAuthorizationController(_: PKPaymentAuthorizationController, didAuthorizePayment payment: PKPayment, handler completion: @escaping (PKPaymentAuthorizationResult) -> Void) {
        handleApplePayPayment(payment) { token, message in
            if let token = token {
                self.token = token
                self.paymentStatus = .success
                completion(PKPaymentAuthorizationResult(status: .success, errors: nil))
            } else {
                self.errorMessage = message
                self.paymentStatus = .failure
                completion(PKPaymentAuthorizationResult(status: .failure, errors: nil))
            }
        }
    }
```
```
private func handleApplePayPayment(_ payment: PKPayment, completion: @escaping ApplePayCompletion) {
        let paymentData = payment.token.paymentData
        debugPrint(payment)
        let apiKey = UserDefaults.isAUUser ? AppConfig.checkoutApiKeyAU : AppConfig.checkoutApiKeyNZ

        let checkoutService = CheckoutAPIService(publicKey: apiKey,
                                                 environment: AppConfig.isDebugMode ? .sandbox : .production)
        checkoutService.createToken(.applePay(.init(tokenData: paymentData))) { result in

             /* THIS BLOCK NEVER GETS CALLED */

            debugPrint("checkout create token result \(result)")
            switch result {
            case let .failure(error):
                debugPrint("Checkout create token failed \(error)")
                var message = ApplePayError.invalidApplePayToken.message
                switch error {
                case .cardValidationError: message = ApplePayError.invalidCard.message
                case .userCancelled: message = ApplePayError.userCancelled.message
                default: ()
                }
                completion(nil, message)
            case let .success(tokenDetails):
                completion(tokenDetails.token, nil)
            }
        }
    }
```

- See Checkout logs produced in the console:
```
Event(typeIdentifier: "card_validator", time: 2024-06-19 07:04:06 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("873f0b53-1582-4621-85bf-9b4bf5d3d087")])

Event(typeIdentifier: "token_requested", time: 2024-06-19 07:04:06 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("873f0b53-1582-4621-85bf-9b4bf5d3d087"), "tokenType": AnyCodable("applepay"), "publicKey": AnyCodable("_KEY REDACTED_")])

Event(typeIdentifier: "token_response", time: 2024-06-19 07:04:07 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["tokenID": AnyCodable("_TOKEN REDACTED_"), "scheme": AnyCodable("VISA"), "tokenType": AnyCodable("applepay"), "httpStatusCode": AnyCodable(201), "publicKey": AnyCodable("_KEY REDACTED_"), "CorrelationId": AnyCodable("873f0b53-1582-4621-85bf-9b4bf5d3d087")])
```

**Comment by mashoba7** [comment]
Thank you for opening this issue, I am having the exact same problem with the following:

Frames SDK v4.3.6

Device: iPad (6th gen)
OS: iPadOS 16.3.1

Device: iPhone 11 Pro
OS: iOS 17.5.1

The callback never gets called, resulting in the Apple Pay sheet displaying `Payment not Completed`, which I guess is due to a timeout issue.

**Comment by ehab-al-cko** [comment]
Hi @ajf-mr @mashoba7,

Thanks for your feedback!

Did you have a chance to check out the Apple Pay implementation and integrate it with the sample app in our current repo? 
It might be helpful to try the sample app using your Apple Pay configurations.

Could you also share your Bundle Identifier with me? I'd like to do some data analysis.

Best,
Ehab

**Comment by ajf-mr** [comment]
Hi @ehab-al-cko,

I ran the example (pod version; for some reason I couldn't change the team in the SPM example) and got this error too:
![Screenshot 2024-06-20 at 8 56 02 AM](https://github.com/checkout/frames-ios/assets/149978177/34278b29-60d4-41ac-a2ce-a796fb98c072)

**Comment by okhan-okbay-cko** [comment]
Hey, failure in the sample project on a simulator is expected, since Apple returns an empty string as an Apple Pay token on simulators. 

However, the callback closure of `createToken(...)` method not being called is something that we couldn't reproduce or detect. Can you share a clean slate project that reproduces it please?

**Comment by okhan-okbay-cko** [comment]
Hey, 

Whilst we are looking at the details of the issue, can you please let me know if this is an intermittent issue or do you keep having the same problem every time you try it?

**Comment by ajf-mr** [comment]
Yes, it always happens (tried on 2 devices), both in sandbox and production environments.

**Comment by okhan-okbay-cko** [comment]
Thanks. We're investigating this at the minute and will come back to you as soon as possible.

**Comment by precious-ossai-cko** [comment]
Hi @ajf-mr, thanks for your patience. Please try this [pre-release version](https://github.com/checkout/frames-ios/tree/pre-release/4.3.7) and confirm if the issue still exists.

**Comment by ajf-mr** [comment]
hi @precious-ossai-cko unfortunately same issue. i tried on sandbox envi, same code as previous. here are the logs printed (keys redacted):

```
<PKPayment: 0x30043aa30; token: <PKPaymentToken: 0x303271800; transactionIdentifier: 666f2a069febf0542a5c8b99122b8364626e236c161cd514891964709c2ddd36; paymentData: 3746 bytes>>
Event(typeIdentifier: "card_validator", time: 2024-07-16 02:05:44 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("0454c863-45ec-466b-8cc6-29e47a55fcd8")])
Event(typeIdentifier: "token_requested", time: 2024-07-16 02:05:44 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["tokenType": AnyCodable("applepay"), "CorrelationId": AnyCodable("0454c863-45ec-466b-8cc6-29e47a55fcd8"), "publicKey": AnyCodable("REDACTED")])
Event(typeIdentifier: "token_response", time: 2024-07-16 02:05:46 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["scheme": AnyCodable("VISA"), "httpStatusCode": AnyCodable(201), "CorrelationId": AnyCodable("0454c863-45ec-466b-8cc6-29e47a55fcd8"), "tokenID": AnyCodable("tok_dgj4lswcihkeleu2mfn6ibq5a4"), "tokenType": AnyCodable("applepay"), "publicKey": AnyCodable("REDACTED")])
_dirhelper_relative_internal: error for path : [2: No such file or directory]
Event(typeIdentifier: "riskDataCollected", time: 2024-07-16 02:05:48 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.debug, properties: ["FpLoad": AnyCodable(17.071291571483016), "FpRequestId": AnyCodable("1721095548003.NMRGXJ"), "FpPublish": AnyCodable(1174.4078332558274), "Block": AnyCodable(698.2872916851193), "Timezone": AnyCodable("Asia/Manila"), "RiskSDKVersion": AnyCodable("3.0.3"), "EventType": AnyCodable("riskDataCollected"), "Total": AnyCodable(1889.7664165124297), "DeviceSessionId": AnyCodable(nil), "FramesMode": AnyCodable(true), "ddTags": AnyCodable("team:prism,service:prism.risk.ios,version:3.0.3,env:sandbox"), "MaskedPublicKey": AnyCodable("pk_sbox_********xzoha2"), "DeviceDataPersist": AnyCodable(nil)])
```

**Comment by okhan-okbay-cko** [comment]
Hey, thanks for the quick response. Can you please reach out to us through a secure channel and your assigned implementation engineer, so that we can have a detailed look at your specific scenario? @ajf-mr

**Comment by okhan-okbay-cko** [comment]
Hi again @ajf-mr, can you please clarify if you're downloading Frames SDK through `CocoaPods` or `SPM`? 

If you're using SPM, can you please point out to [this branch](https://github.com/checkout/frames-ios/tree/pre-release/4.3.7-2) and see if the issue still persists?

**Comment by ajf-mr** [comment]
hey @okhan-okbay-cko tried it and that worked. i see an error log for Risk but the completion block was called and the test payment went through.

```
Event(typeIdentifier: "card_validator", time: 2024-07-24 02:35:24 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("175f322f-5317-4363-adcb-d0b02d615709")])
Event(typeIdentifier: "token_requested", time: 2024-07-24 02:35:24 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("175f322f-5317-4363-adcb-d0b02d615709"), "tokenType": AnyCodable("applepay"), "publicKey": AnyCodable("REDACTED")])
Event(typeIdentifier: "token_response", time: 2024-07-24 02:35:26 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["scheme": AnyCodable("VISA"), "publicKey": AnyCodable("REDACTED"), "tokenID": AnyCodable("tok_tltjs7oybeiujitukuky7vcqpi"), "CorrelationId": AnyCodable("175f322f-5317-4363-adcb-d0b02d615709"), "httpStatusCode": AnyCodable(201), "tokenType": AnyCodable("applepay")])
Publish data under protocol called
Event(typeIdentifier: "riskDataPublishFailure", time: 2024-07-24 02:35:29 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.debug, properties: ["ErrorReason": AnyCodable("persistFpData"), "Timezone": AnyCodable("Asia/Manila"), "MaskedPublicKey": AnyCodable("pk_sbox_********xzoha2"), "EventType": AnyCodable("riskDataPublishFailure"), "FramesMode": AnyCodable(true), "RiskSDKVersion": AnyCodable("3.0.3"), "Block": AnyCodable(1749.375875107944), "FpPublish": AnyCodable(0.0), "DeviceDataPersist": AnyCodable(nil), "ErrorMessage": AnyCodable("The operation couldn’t be completed. (Risk.APIServiceError error 0.)"), "ddTags": AnyCodable("team:prism,service:prism.risk.ios,version:3.0.3,env:sandbox"), "ErrorType": AnyCodable("Error"), "FpLoad": AnyCodable(0.0), "Total": AnyCodable(1749.375875107944)])
Event(typeIdentifier: "risk_sdk_completion", time: 2024-07-24 02:35:29 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("175f322f-5317-4363-adcb-d0b02d615709")])
```

**Comment by okhan-okbay-cko** [comment]
Hey, thanks for your collaboration and quick responses! @ajf-mr 

We'll have this issue unblocked after the latest developments that live on [this branch](https://github.com/checkout/frames-ios/tree/feature/risk-sdk-timeout-recovery). Can you please point to that branch and let us know if you can get the token response successfully so that we go forward with the release?

Many thanks, 
Okhan

**Comment by ajf-mr** [comment]
@okhan-okbay-cko confirming that i'm getting the token response when pointing to that branch.

```
Event(typeIdentifier: "card_validator", time: 2024-08-05 23:32:54 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("784deff1-5c32-4ff5-a5fc-9515b2a5672a")])
Event(typeIdentifier: "token_requested", time: 2024-08-05 23:32:54 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("784deff1-5c32-4ff5-a5fc-9515b2a5672a"), "publicKey": AnyCodable("REDACTED"), "tokenType": AnyCodable("applepay")])
tcp_input [C14.1.1.1:3] flags=[R.] seq=4083917072, ack=3141849554, win=59 state=LAST_ACK rcv_nxt=4083917072, snd_una=3141849554
tcp_input [C14.1.1.1:3] flags=[R] seq=4083917072, ack=0, win=0 state=CLOSED rcv_nxt=4083917072, snd_una=3141849554
tcp_input [C14.1.1.1:3] flags=[R] seq=4083917072, ack=0, win=0 state=CLOSED rcv_nxt=4083917072, snd_una=3141849554
Event(typeIdentifier: "token_response", time: 2024-08-05 23:32:56 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["scheme": AnyCodable("VISA"), "httpStatusCode": AnyCodable(201), "CorrelationId": AnyCodable("784deff1-5c32-4ff5-a5fc-9515b2a5672a"), "tokenType": AnyCodable("applepay"), "publicKey": AnyCodable("REDACTED"), "tokenID": AnyCodable("tok_tf66hmqvi7eunfd3zyy5gc3ize")])
_dirhelper_relative_internal: error for path : [2: No such file or directory]
Event(typeIdentifier: "riskDataCollected", time: 2024-08-05 23:32:58 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.debug, properties: ["Timezone": AnyCodable("Asia/Manila"), "DeviceDataPersist": AnyCodable(nil), "MaskedPublicKey": AnyCodable("pk_sbox_********xzoha2"), "FpPublish": AnyCodable(1201.4741250313818), "ddTags": AnyCodable("team:prism,service:prism.risk.ios,version:3.0.2,env:sandbox"), "FpRequestId": AnyCodable("1722900778223.mTpbL9"), "Block": AnyCodable(693.9647081308067), "EventType": AnyCodable("riskDataCollected"), "RiskSDKVersion": AnyCodable("3.0.2"), "DeviceSessionId": AnyCodable(nil), "FpLoad": AnyCodable(15.303833410143852), "Total": AnyCodable(1910.7426665723324), "FramesMode": AnyCodable(true)])
Event(typeIdentifier: "riskDataPublished", time: 2024-08-05 23:32:58 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.debug, properties: ["Block": AnyCodable(693.9647081308067), "Timezone": AnyCodable("Asia/Manila"), "FpRequestId": AnyCodable("1722900778223.mTpbL9"), "EventType": AnyCodable("riskDataPublished"), "RiskSDKVersion": AnyCodable("3.0.2"), "MaskedPublicKey": AnyCodable("pk_sbox_********xzoha2"), "FpLoad": AnyCodable(15.303833410143852), "DeviceDataPersist": AnyCodable(435.6047082692385), "DeviceSessionId": AnyCodable("tok_tf66hmqvi7eunfd3zyy5gc3ize"), "FpPublish": AnyCodable(1201.4741250313818), "FramesMode": AnyCodable(true), "Total": AnyCodable(2346.347374841571), "ddTags": AnyCodable("team:prism,service:prism.risk.ios,version:3.0.2,env:sandbox")])
Event(typeIdentifier: "risk_sdk_completion", time: 2024-08-05 23:32:58 +0000, monitoringLevel: CheckoutEventLoggerKit.MonitoringLevel.info, properties: ["CorrelationId": AnyCodable("784deff1-5c32-4ff5-a5fc-9515b2a5672a")])
```

**Comment by okhan-okbay-cko** [comment]
Hey,

Please use version 4.3.7 that's just been published now.

If the problem persists, please let us know.

Thank you for reporting and helping us to resolve the issue @ajf-mr,
Okhan
